front-end/src/components/Post/Democracy.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { web3Accounts, web3FromSource, web3Enable } from '@polkadot/extension-dapp';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
+
+import { UserDetailsContext } from '../../context/UserDetailsContext';
+import { NotificationContext } from '../../context/NotificationContext';
+import { NotificationStatus } from '../../types';
+import SecondProposal from './SecondProposal';
+import VoteRefrendum from './VoteRefrendum';
+
+interface Props {
+	isProposal?: boolean
+	isReferendum?: boolean
+	onchainId?: number | null | undefined
+}
+
+const WS_PROVIDER = process.env.REACT_APP_WS_PROVIDER || 'wss://kusama-rpc.polkadot.io';
+const APPNAME = process.env.REACT_APP_APPNAME || 'polkassembly';
+
+const Democracy = ({ isProposal, isReferendum, onchainId }: Props) => {
+	const currentUser = useContext(UserDetailsContext);
+
+	const [api, setApi] = useState<ApiPromise>();
+	const [apiReady, setApiReady] = useState(false);
+	const { queueNotification } = useContext(NotificationContext);
+
+	useEffect(() => {
+		async function connect() {
+			const provider = new WsProvider(WS_PROVIDER);
+			const apiResult = await ApiPromise.create({ provider });
+
+			setApi(apiResult);
+			apiResult.isReady.then(() => {
+				setApiReady(true);
+				console.log('API ready');
+			});
+		}
+
+		connect().catch((error) => {
+			queueNotification({
+				header: 'Failed!',
+				message: error.message,
+				status: NotificationStatus.ERROR
+			});
+		});
+	}, []);
+
+	const getLinkedAccount = async (): Promise<InjectedAccountWithMeta | undefined> => {
+		const extensions = await web3Enable(APPNAME);
+
+		if (extensions.length === 0) {
+			queueNotification({
```

Comment:
This shouldn't be a notification, it should rather be an error message in place of the voting buttons.

Reply:
I actually never saw this notification, when I disable the extension there's nothing specific happening other than the voting buttons flashing.

Reply:
Move to error message:

![Screenshot from 2020-02-09 12-59-16](https://user-images.githubusercontent.com/874046/74098252-00194b00-4b3c-11ea-9a09-4230c70b42fd.png)

Reply:
should the voting button be displayed with this message or should it be hidden? any comments on design, color of message.